GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -186,8 +187,8 @@ private CommitKind CommitKind
             {
                 _commitKind = value;
 
-                modifyCommitMessageButton.Visible = _useFormCommitMessage && CommitKind != CommitKind.Normal;
-                bool messageCanBeChanged = _useFormCommitMessage && CommitKind == CommitKind.Normal;
+                modifyCommitMessageButton.Visible = _useFormCommitMessage && CommitKind is not CommitKind.Normal and not CommitKind.Amend;
+                bool messageCanBeChanged = _useFormCommitMessage && CommitKind is CommitKind.Normal or CommitKind.Amend;
```

Comment:
❓ similar as for line 461:
```suggestion
                modifyCommitMessageButton.Visible = _useFormCommitMessage && CommitKind is not (CommitKind.Normal or CommitKind.Amend);
                bool messageCanBeChanged = _useFormCommitMessage && CommitKind is (CommitKind.Normal or CommitKind.Amend);
```